orc8r/tools/helm/promote.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# promote.sh copies specified artifacts from one repo to another
+
+set -e -o pipefail
+
+promote_artifact () {
+  ARTIFACT="$1"
+  curl -X POST -u "$HELM_CHART_MUSEUM_USERNAME":"$HELM_CHART_MUSEUM_TOKEN" --fail \
+   "$HELM_CHART_MUSEUM_API_URL/copy/$HELM_CHART_MUSEUM_ORIGIN_REPO/$ARTIFACT?to=/$HELM_CHART_MUSEUM_DEST_REPO/$ARTIFACT"
+} 
+
+get_artifact () {
+  ARTIFACT="$1"
+  curl -o /dev/null -s -w "%{http_code}" \
+   "$HELM_CHART_ARTIFACTORY_URL/$HELM_CHART_MUSEUM_ORIGIN_REPO/$ARTIFACT"
+} 
+
+usage() {
+  echo "Supply at least one artifact to promote: $0 ARTIFACT_PATH"
+  exit 2
+}
+
+exitmsg() {
+  echo "$1"
+  exit 1
+}
+
+# Check if the required args and env-vars present
+[ $# -eq 0 ] && usage
+
+if [[ -z $HELM_CHART_ARTIFACTORY_URL ]]; then
+  HELM_CHART_ARTIFACTORY_URL="https://artifactory.magmacore.org:443/artifactory"
+fi
+
+if [[ -z $HELM_CHART_MUSEUM_ORIGIN_REPO ]]; then
+  HELM_CHART_MUSEUM_ORIGIN_REPO="helm-test"
+fi
+
+if [[ -z $HELM_CHART_MUSEUM_DEST_REPO ]]; then
+  HELM_CHART_MUSEUM_DEST_REPO="helm-prod"
+fi
+
+if [[ -z $HELM_CHART_MUSEUM_USERNAME ]]; then
+  exitmsg "Environment variable HELM_CHART_MUSEUM_USERNAME must be set"
+fi
+
+if [[ -z $HELM_CHART_MUSEUM_TOKEN ]]; then
+  exitmsg "Environment variable HELM_CHART_MUSEUM_TOKEN must be set"
+fi
+
+# Trim last backslash if exists
+# shellcheck disable=SC2001
+HELM_CHART_ARTIFACTORY_URL="$(echo "$HELM_CHART_ARTIFACTORY_URL" | sed 's:/$::')"
+# shellcheck enable=SC2001
+
+# Verify existence of the helm repo
+set +e +o pipefail
+RESPONSE_CODE_REPO="$(curl -o /dev/null -s -w "%{http_code}"  "$HELM_CHART_ARTIFACTORY_URL/$HELM_CHART_MUSEUM_ORIGIN_REPO/")"
+if [ "$RESPONSE_CODE_REPO" != "200" ]; then
+  exitmsg "There was an error connecting to the artifactory repository $HELM_CHART_MUSEUM_ORIGIN_REPO"
+fi
+set -e -o pipefail
+
+# Form API URL
+HELM_CHART_MUSEUM_API_URL="$HELM_CHART_ARTIFACTORY_URL/api"
+
+# iterate through artifacts to promote
+for artifact in "$@"
+do
+    set +e +o pipefail
```

Comment:
I think there might be a way to get curl to *not* exit with error on bad request. That would be preferable to this verbose option toggling. Lmk if you're able to get that to work!

Reply:
If not, encapsulate every usage of the option toggle within a function, for added readability. E.g. "set +e ..., get_artifact, set -e ..." encapsulated in a single function